ProtocolHandler.pm

```diff
@@ -72,7 +72,14 @@ sub getFormatForURL {
 # }
 
 sub canSeek { 0 }
-sub canDirectStreamSong { 0 }
+
+# allow players that can handle HTTPS to decide
+sub canDirectStream { 
+	my $class = shift;
+	my $client = shift;
+	my $url = shift;
```

Comment:
Isn't the third parameter the `$song` object? Other implementations look like this:

```
sub canDirectStreamSong {
	my ( $class, $client, $song ) = @_;
	
	# We need to check with the base class (HTTP) to see if we
	# are synced or if the user has set mp3StreamingMethod
	return $class->SUPER::canDirectStream($client, $song->streamUrl(), $class->getFormatForURL());
}
```

I'll give that a try right now.

Reply:
Your version actually fails with
```
[19-04-28 20:27:28.9086] Slim::Networking::IO::Select::__ANON__ (131) Error: Select task failed calling Slim::Networking::Slimproto::client_readable: Can't locate object method "streamUrl" via package "Slim::Schema::RemoteTrack" at /Users/mh/Dropbox/Squeezebox/git/server/Plugins/RadioParadise/ProtocolHandler.pm line 359.
; fh=IO::Socket::INET=GLOB(0x7f926b56e8e0)
```

Reply:
Oh... that's what the other changes are for :-). Never mind my previous comment.

Reply:
There are two canDirectStream methods. On that applies to a player $client and that one has  $client, $url and $song and the other one applies to a protocol handler and has $class, $client and $url. Here is what I have in Squeezebox2.pm
[CODE]
sub canDirectStream {
	my $client = shift;
	my $url = shift;
	my $song = shift;

	my $handler = Slim::Player::ProtocolHandlers->handlerForURL($url);

	if ($song && $handler && $handler->can("canDirectStreamSong")) {
		return $handler->canDirectStreamSong($client, $song);
	} elsif ($handler && $handler->can("canDirectStream")) {
		return $handler->canDirectStream($client, $url);
	}

	return undef;
}
[/CODE]